### PR TITLE
Implement api top books borrow

### PR DESCRIPTION
**Pull request by at-trampham**
## Description
api list top book borrowed
## Reference Requirements
- Product Backlog ID: PBI-36
- Task ID: #36.5
## Migrations
YES

## Impacted Areas in Application
List general components of the application that this PR will affect:

-
## SQL query and `explain`
select `id`, `title`, `rating`, `total_rating`, `picture`, (select count(*) from `borrows` where `books`.`id` = `borrows`.`book_id` and `borrows`.`deleted_at` is null) as `borrowed` from `books` where `books`.`deleted_at` is null order by `borrowed` desc limit 20 offset 0
image

- Refer to [Using Explain with MySQL](https://viblo.asia/p/su-dung-explain-de-toi-uu-cau-lenh-mysql-BYjv44gmvxpV)

### CODE REVIEW

File: app/Http/Controllers/Api/BookController.php [hunk 1]
```diff
@@ -95,4 +95,27 @@ public function topBooksReview()
...
+        $fields =  [
+            'id',
+            'title',
+            'rating',
+            'total_rating',
+            'picture'
+        ];
+
+        $topBooks = Book::select($fields)
+                    ->withCount('borrows AS borrowed')
```
Comment (at-minhdao) [inline app/Http/Controllers/Api/BookController.php]: fix space nha :D

**Review by at-minhdao — COMMENTED** [review]
check cmt

**Review by at-minhdao — COMMENTED** [review]
LGTM!

File: routes/api.php [hunk 2]
```diff
@@ -19,6 +19,7 @@
 
 Route::group(['namespace' => 'Api'], function(){
     Route::get('categories', 'CategoryController@index');
+    Route::get('books/top-borrow', 'BookController@topBooksBorrow');
```
Comment (at-xinhnguyen) [inline routes/api.php]: `endpoint` phải là danh từ số nhiều em, follow theo chuẩn `restful api`

**Comment by at-xinhnguyen** [comment]
please attach list query and explain for them in description

File: app/Http/Controllers/Api/BookController.php [hunk 3]
```diff
@@ -95,4 +95,27 @@ public function topBooksReview()
...
+            'id',
+            'title',
+            'rating',
+            'total_rating',
+            'picture'
+        ];
+
+        $topBooks = Book::select($fields)
+                        ->withCount('borrows AS borrowed')
+                        ->orderBy('borrowed', 'desc')
```
Comment (at-xinhnguyen) [inline app/Http/Controllers/Api/BookController.php]: SQL keyword should be uppercase

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**